legacy/src/main/java/org/opensearch/sql/legacy/cursor/DefaultCursor.java

```diff
@@ -70,29 +91,67 @@ public class DefaultCursor implements Cursor {
   /** To get next batch of result */
   private String scrollId;
 
+  /** To get Point In Time */
+  private String pitId;
+
+  /** To get next batch of result with search after api */
+  public SearchSourceBuilder searchSourceBuilder;
+
+  /** To get last sort values * */
+  private Object[] sortFields;
+
   /** To reduce the number of rows left by fetchSize */
   @NonNull private Integer fetchSize;
 
   private Integer limit;
 
+  /**
+   * {@link NamedXContentRegistry} from {@link SearchModule} used for construct {@link QueryBuilder}
+   * from DSL query string.
+   */
+  private static final NamedXContentRegistry xContentRegistry =
+      new NamedXContentRegistry(
+          new SearchModule(Settings.EMPTY, Collections.emptyList()).getNamedXContents());
+
   @Override
   public CursorType getType() {
     return type;
   }
 
   @Override
   public String generateCursorId() {
-    if (rowsLeft <= 0 || Strings.isNullOrEmpty(scrollId)) {
+    if (rowsLeft <= 0 || isCursorValid()) {
       return null;
     }
     JSONObject json = new JSONObject();
     json.put(FETCH_SIZE, fetchSize);
     json.put(ROWS_LEFT, rowsLeft);
     json.put(INDEX_PATTERN, indexPattern);
-    json.put(SCROLL_ID, scrollId);
     json.put(SCHEMA_COLUMNS, getSchemaAsJson());
     json.put(FIELD_ALIAS_MAP, fieldAliasMap);
-    return String.format("%s:%s", type.getId(), encodeCursor(json));
+    if (LocalClusterState.state().getSettingValue(SQL_PAGINATION_API_SEARCH_AFTER)) {
```

Comment:
I see this branching logic in many place. And it is making the code more complicated. Can't we implement the logic for PIT in a separate class?

Reply:
I think for now both the approaches are supported and it is temporary till we get benchmark both the performances and remove scroll support at some point. I can create an issue for this to track - https://github.com/opensearch-project/sql/issues/2945

Reply:
I don't think this is a blocker. I think supporting both the approaches with default to PIT was the plan from the beginning as user can still set it in the setting. At least I don't see anything called out in the hand-off doc. Let me know if you think this is a blocker.

Reply:
Lets make sure we have something to track this

Reply:
Yes I have created the issue already https://github.com/opensearch-project/sql/issues/2945